File: Software/travelportal_backend/RewardService/src/main/java/at/technikumwien/rewardservice/Model/Blog.java
package at.technikumwien.rewardservice.Model;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.stereotype.Component;


@Component
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Blog {

    private Long id;

    private String title;

    private String blogText;

    private Author author;

    public Blog(String title, String blogText, Author author)
    {
        this(null,title,blogText,author);
    }

    public Blog(String title, Author author)
    {
        this(null,title,null, author);
    }

}

File: Software/travelportal-frontend/src/app/shared/Blog.ts
export interface Blog {
  id: number,
  title: string,
  blogText: string,
  author: Author,
  attraction: Attraction
}

export interface Author{
  id: number,
  firstname: string,
  lastname: string
}

export interface Attraction{
  id: number,
  name: string,
  description: string
}
